compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -139,6 +139,15 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
     try topLevelSubType(tp1, tp2)
     finally useNecessaryEither = saved
 
+  /** Use avoidance to get rid of wildcards in constraint bounds if
+   *  we are doing a neccessary comparison, or the mode is TypeVarsMissContext.
+   *  The idea is that under either of these conditions we are not interested
+   *  in creating a fresh type variable to replace the wildcard. I verified
```

Comment:
Yes, it probably should. Perhaps no GADT tests failed, because the interaction of GADTs and wildcards is undertested.

As a side remark, I'm not sure when exactly we should consider `useNecessaryEither` in separation from the GADT mode. The name doesn't help either in reminding that we should do both. Perhaps we should hide this condition behind a definition and rename `useNecessaryEither`?

Reply:
I believe 12677 is the test case that fails otherwise.

Reply:
Not sure about GADT inference. Does it even come up?